updateHostsFile.py

```diff
@@ -1482,12 +1482,17 @@ def get_file_by_url(url):
         format we have to encode or decode data before parsing it to UTF-8.
     """
 
-    try:
-        f = urlopen(url)
-        soup = BeautifulSoup(f.read(), "lxml").get_text()
-        return "\n".join(list(map(domain_to_idna, soup.split("\n"))))
-    except Exception:
-        print("Problem getting file: ", url)
+    while retries:
+        try:
+            with urlopen(url) as f:
+                soup = BeautifulSoup(f.read(), "lxml").get_text()
+                return "\n".join(list(map(domain_to_idna, soup.split("\n"))))
+        except Exception as e:
+            if 'failure in name resolution' in str(e):
+                print('No internet connection! Retrying in {} seconds'.format(delay))
+                time.sleep(delay)
+                retries -= 1
+            return print("Problem getting file: ", url)
```

Comment:
@codeswhite, you are not retrying with your `return` statement.

1. This return statement combined with the `print()` function might be obscure for future python contributors. Let's keep it in the same format as before :wink: 
2. This return statement should be outside of the `while` statement if we want to retry.

In the current form and in my understanding, you are not retrying.
Instead, you are handling the resolution issue, waiting for 10 seconds and then return `None`. 

Isn't it better to add a new exception handler? Indeed, we could have 2 handlers. A more general `Exception` for everything we don't necessarily want to control and the `urllib.error.URLError` exception with the `failure in name resolution` message which will handler our delay.

Please let me know if I missed something or if my understanding late at night (Berlin Time) is wrong.